File: Backend/src/Models/Hospital.js
const mongoose = require("mongoose");
const { Schema } = mongoose;

const HospitalScehma = new Schema({
    name: {
        type: String,
        required: true,
    },
    email: {
        type: String,
        required: true,
    },
    password: {
        type: String,
        required: true,
    },
    phoneNo: {
        type: String,
        required: true,
    },
    address: {
        type: String,
        required: true,
    },
    city: {
        type: String,
        required: true,
    },
    state: {
        type: String,
        required: true,
    },
    pincode: {
        type: String,
        required: true,
    },
    latitude: {
        type:Schema.Types.Decimal128,
        required: true,
    },
    longitude: {
        type:Schema.Types.Decimal128,
        required: true,
    },
    capacity:{
        type:Number,
        default:0,
        required:true,
    },
    isActive:{
        type:Boolean,
        default:true,
        required:true,
    },
    remCapacity:{
        type:Number,
        default:0,
        required:true,
    },
    info:{
        type:String,
        default:"",
        required:true,
    },
    date: {
        type: Date,
        required: true,
        default: Date.now,
    },
})

const Hospital = mongoose.model("hospital", HospitalScehma);
module.exports = { Hospital };

File: App/app/src/main/java/com/example/emergencyalert/hospitals/dto/Hospital.kt
package com.example.emergencyalert.hospitals.dto

import android.health.connect.datatypes.DistanceRecord
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Hospital (
    @SerialName("_id") val id: String,
    val name: String,
    val email: String,
    @SerialName("phoneNo") val phoneNo: String,
    val address: String,
    val city: String,
    val state: String,
    val pincode: String,
    val latitude: Latitude,
    val longitude: Longitude,
    val info: String,
    val date: String,
    val capacity: Int,
    val isActive: Boolean,
    val remCapacity: Int,
    val distance:Double?
)

@Serializable
data class Longitude(@SerialName("\$numberDecimal") val value: String)

@Serializable
data class Latitude(@SerialName("\$numberDecimal") val value: String)
